From 54743b5ab981d686b885c3da639d4ed6cc995e8b Mon Sep 17 00:00:00 2001
From: Jiazhi Huang <huangjia2hi@163.com>
Date: Tue, 11 Aug 2026 17:37:44 +0800
Subject: [PATCH 13229/15333] hwmon: (nct6775) Add NCT5585D to list of
 supported chips

NCT5585D uses the same chip ID (0xd428) as NCT6798D and is fully
compatible with it. Tested on an ASUS PRIME H610M-D motherboard.

Signed-off-by: Jiazhi Huang <huangjia2hi@163.com>
Link: https://lore.kernel.org/r/20260811093744.12199-3-huangjia2hi@163.com
Signed-off-by: Guenter Roeck <linux@roeck-us.net>
---
 Documentation/hwmon/nct6775.rst  | 8 ++++++++
 drivers/hwmon/nct6775-core.c     | 1 +
 drivers/hwmon/nct6775-platform.c | 2 +-
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/Documentation/hwmon/nct6775.rst b/Documentation/hwmon/nct6775.rst
index 6725f8ff98e0..41b21bed7e27 100644
--- a/Documentation/hwmon/nct6775.rst
+++ b/Documentation/hwmon/nct6775.rst
@@ -96,6 +96,14 @@ Supported chips:
 
     Datasheet: Available from Nuvoton upon request
 
+  * Nuvoton NCT5585D (compatible to NCT6798D)
+
+    Prefix: 'nct6798'
+
+    Addresses scanned: ISA address retrieved from Super I/O registers
+
+    Datasheet: Not available
+
   * Nuvoton NCT6796D-S/NCT6799D-R
 
     Prefix: 'nct6799'
diff --git a/drivers/hwmon/nct6775-core.c b/drivers/hwmon/nct6775-core.c
index 94482c8092cd..759ea3d1286e 100644
--- a/drivers/hwmon/nct6775-core.c
+++ b/drivers/hwmon/nct6775-core.c
@@ -33,6 +33,7 @@
  *                                           (0xd451)
  * nct6798d    14      7       7       2+6    0xd428 0xc1    0x5ca3
  *                                           (0xd429)
+ * nct5585d    14      7       7       2+6    0xd428 0xc1    0x5ca3
  * nct6796d-s  18      7       7       6+2    0xd801 0xc1    0x5ca3
  * nct6799d-r  18      7       7       6+2    0xd802 0xc1    0x5ca3
  *
diff --git a/drivers/hwmon/nct6775-platform.c b/drivers/hwmon/nct6775-platform.c
index 1975399ac440..d394376e4306 100644
--- a/drivers/hwmon/nct6775-platform.c
+++ b/drivers/hwmon/nct6775-platform.c
@@ -34,7 +34,7 @@ static const char * const nct6775_sio_names[] __initconst = {
 	[nct6795] = "NCT6795D",
 	[nct6796] = "NCT6796D",
 	[nct6797] = "NCT6797D",
-	[nct6798] = "NCT6798D",
+	[nct6798] = "NCT6798D/NCT5585D",
 	[nct6799] = "NCT6796D-S/NCT6799D-R",
 };
 
-- 
2.55.0

